### PR TITLE
Fixed JS api url sanitation.

### PR DESCRIPTION
**Pull request by boocmp**
Resolves https://github.com/brave/brave-browser/issues/41064

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by goodov — APPROVED** [review]
`chromium_src` lgtm

File: chromium_src/content/browser/renderer_host/clipboard_host_impl.cc [hunk 1]
```diff
@@ -24,8 +24,12 @@ std::u16string Sanitize(content::ContentBrowserClient* client,
 
   const GURL url(data);
   if (url.is_valid() && !url.is_empty()) {
```
Comment (fmarier) [inline chromium_src/content/browser/renderer_host/clipboard_host_impl.cc:26]: Let's make sure we only ever modify HTTP/HTTPS URLs:
```suggestion
  if (url.is_valid() && !url.is_empty() && url.SchemeIsHTTPOrHTTPS()) {
```

File: browser/url_sanitizer/url_sanitizer_browsertest.cc [hunk 2]
```diff
@@ -161,48 +162,51 @@ class URLSanitizerTestBase : public InProcessBrowserTest {
...
-    check("test_3", false);
-    check("test_4", false);
-    check("test_5", false);
+    check("test_1", false, unsanitized);
+    check("test_2", false, unsanitized);
+    check("test_3", false, unsanitized);
+    check("test_4", false, unsanitized);
+    check("test_5", false, unsanitized);
+    check("test_sanitizable_password", should_sanitize, unsanitized);
+    check("test_not_sanitizable_password", should_sanitize, "Pa$$w0rd");
```
Comment (fmarier) [inline browser/url_sanitizer/url_sanitizer_browsertest.cc:209]: Could we also add these test cases:
- `A:^C,D`
- `Ftp://Example.Com/?si=12345`

Neither of them should be sanitized. One is the test case from the password manager bug. The other is scheme we don't support.

**Review by fmarier — CHANGES_REQUESTED** [review]

**Review by fmarier — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.72.34 Chromium: 129.0.6668.59 (Official Build) nightly (64-bit)
-- | --
Revision | 74bf470a0291ea3719f1a75af066ee10b7057dbd
OS | Windows 11 Version 23H2 (Build 22631.4169)
```

Reproduced the original issue using `1.72.31 Chromium: 129.0.6668.59` with the STR/Cases outlined via https://github.com/brave/brave-browser/issues/41064#issue-2525717014 & https://fmarier.github.io/brave-testing/clipboard.html.

Once the above was reproduced, ensured that the above cases were correctly working and the correct text/strings/links were being copied as per the following:

* visited vault.bitwarden.com and created a new entry with `ASDASD:asd^ASD,ASD` as the password
* once the new entry was created, ensured that copying via the icon copied/pasted `ASDASD:asd^ASD,ASD`
* visited https://fmarier.github.io/brave-testing/clipboard.html and ensured that `A:^C,D` was copied/pasted

**Pull request merged**